### ISSUE TITLE
Intermittent 'Bad Gateway' Error using Cloudflare Tunnels

### ISSUE THREAD

**Issue by some-guy-23** [autobrr/autobrr#806]
**Version**
1.23.0

**Describe the bug**
Using Cloudflare Tunnels as a reverse proxy, there seems to be an issue connecting to autobrr. I have my host binding set to `0.0.0.0` and am able to access the service fine on my local network. However, when accessing via Cloudflare tunnel, the service occasionally starts to return a 502 Bad Gateway error. It will eventually go back to working, without any configuration changes. I've tried tweaking the "host" header (from Cloudflare public hostname config) sent as well to be a local IP / port, and for a while thought this fixed the issue, but then started getting the error again.

**To Reproduce**
1. Bind autobrr to host `0.0.0.0`
2. Setup a [Cloudflare Tunnel agent](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/tunnel-guide/remote/), pointing to an accessible local IP / port of autobrr
3. Attempt to access the site
4. See frequent 502 Bad Gateway errors

**Expected behavior**
The service to load consistently, without Bad Gateway error.

**Comment by zze0s** [comment]
Hey! That's an interesting issue.

Can you set log level to TRACE and see if it reaches autobrr at all? Do you have any reverse proxies behind the tunnel that could interfere?

I don't see how it would be an issue with autobrr itself, rather than your surrounding setup, but with some logs you'll see if it can reach it or not.

**Comment by some-guy-23** [comment]
Thanks for your response! Something occurred to me after your question, I have multiple cloudflared agents installed, one of which is in a Docker container. Cloudflare docs say it will randomly choose which agent to route through. So I disabled all but one at a time, and narrowed it down to an issue with the Docker container agent. From there I found that it was the local firewall on the autobrr host machine that was blocking the connection from Docker machine.

Sorry for the confusion, was all on my end, I'll close this issue!
